docs/proposals/hpa/federated-hpa-v2.md

```diff
@@ -0,0 +1,785 @@
+---
+title: Federated HPA
+authors:
+- "@jwcesign" # Authors' GitHub accounts here.
+reviewers:
+- TBD
+approvers:
+- TBD
+
+creation-date: 2023-02-14
+
+---
+
+# Federated HPA
+## Summary
+HPA is a widely used approach to improve the stability of apps dealing with the burst of requests and resource utilization. We can run our apps on multiple clusters as easily as a single cluster with Karmada, but Karmada lacks HPA support now. 
+This proposal introduces a new component FederatedHPA to Karmada and enables Karmada to scale workloads across clusters with different policies. It also tries to minimize the user experience differences between HPA in a single cluster and multi-clusters.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.
+-->
+
+If users deploy the workloads in multi-clusters, they may want to scale the workloads in different clusters with different policies. For example, the workloads in the cluster with high priority may need to be scaled first before the clusters with low priority.
+
+### Goals
+1. Bring HPA from single cluster to multi-clusters.
+1. Compatible with the HPA-related resources in a single cluster.
+1. Tolerate the disaster of member clusters or Karmada control plane.
+1. It is better to integrate well with scenarios such as cloud burst.
+1. It is better to support both Kubernetes HPA and customized HPA.
+1. Minimize control plane overhead when scaling workloads across clusters, and maximize the utilization of member clusters' scaling capabilities. 
+1. Having rich scaling policies, users can enable different scenarios with different configurations.
```

Comment:
> Having rich scaling policies

I'm curious, in what scenario would a user not use a custom HPA directly? Instead, they want to use the policies in Federated-HPA?

Reply:
For example, scaling up the replicas in the clusters with high priority.